tests/instancetype_test.go

```diff
@@ -397,6 +398,40 @@ var _ = Describe("[crit:medium][vendor:cnv-qe@redhat.com][level:component][sig-c
 			Expect(vmi.Annotations).To(HaveKeyWithValue("preferred-annotation-1", "1"))
 			Expect(vmi.Annotations).To(HaveKeyWithValue("preferred-annotation-2", "2"))
 		})
+
+		FIt("[test_id:CNV-10328] should apply preference with PreferredStorageClassName to VM", func() {
+
+			StorageClassName := getAvailableStorageClass(virtClient)
+
+			clusterPreference := newVirtualMachineClusterPreference()
+			clusterPreference.Spec.Volumes = &instancetypev1beta1.VolumePreferences{
+				PreferredStorageClassName: StorageClassName,
+			}
+			clusterPreference, err := virtClient.VirtualMachineClusterPreference().
+				Create(context.Background(), clusterPreference, metav1.CreateOptions{})
+			Expect(err).ToNot(HaveOccurred())
+
+			dataVolume := libdv.NewDataVolume(
+				libdv.WithRegistryURLSource(cd.DataVolumeImportUrlForContainerDisk(cd.ContainerDiskCirros)),
+				libdv.WithPVC(libdv.PVCWithStorageClass(StorageClassName)),
+			)
+			vm := libvmi.NewVirtualMachine(
+				libvmi.New(
+					libvmi.WithDataVolume("disk0", dataVolume.Name),
+					libvmi.WithResourceMemory("100M"),
+				),
+				libvmi.WithDataVolumeTemplate(dataVolume),
+			)
+			vm.Spec.Preference = &v1.PreferenceMatcher{
+				Name: clusterPreference.Name,
+			}
+
+			vm, err = virtClient.VirtualMachine(testsuite.GetTestNamespace(vm)).Create(context.Background(), vm)
+			Expect(err).ToNot(HaveOccurred())
+			_ = virtClient.VirtualMachine(vm.Namespace).Start(context.Background(), vm.Name, &v1.StartOptions{})
```

Comment:
I don't think it is necessary to start the VM. The preference is applied in the VM mutating webhook during the creation.